src/sort.rs

```diff
@@ -0,0 +1,121 @@
+//! # Sort
+//!
+//! Contains Sorter and SortOrder that use command line arguments to sort images
+
+// Only because of SortOrder, as a result of Clap bug/restriction that only permits bare variants
+// inside of arg!_enum macro call
+#![allow(missing_docs)]
+
+use core::cmp::Ordering;
+use fs_extra::dir::get_size;
+use std::ffi::OsStr;
+use std::path::PathBuf;
+use std::time::SystemTime;
+
+/// A structure that takes the SortOrder from command line arguments
+/// and reverse and sorts pathes based on original criteria
+// Chosen over functions, due to it being more adaptable in the future
+pub struct Sorter {
+    /// Order to sort the files
+    sort_order: SortOrder,
+    /// Whether or not to reverse the order of sorting
+    reverse: bool,
+}
+
+impl Sorter {
+    /// Create sorter with command line arguments SortOrder and reverse
+    pub fn new(sort_order: SortOrder, reverse: bool) -> Self {
+        Self {
+            sort_order,
+            reverse,
+        }
+    }
+
+    /// Change the order of sorting, this function doesn't sort upon change
+    pub fn set_order(&mut self, sort_order: SortOrder) {
+        self.sort_order = sort_order;
+    }
+
+    /// Change whether or not to reverse the sorting of images
+    pub fn set_reverse(&mut self, reverse: bool) {
+        self.reverse = reverse;
+    }
+
+    /// Sorts the images based on sort_order, reverses if necessary
+    pub fn sort<'a>(&self, pathes: &'a mut Vec<PathBuf>) -> &'a mut Vec<PathBuf> {
+        pathes.sort_by(|a, b| self.sort_order.file_compare(&a, &b));
+        if self.reverse {
+            pathes.reverse();
+        }
+        pathes
```

Comment:
It this a deliberate spelling choice because `paths` conflicts with something?

Reply:
Yeah, that's a typo...